File: app/src/main/java/mchehab/com/knearestneighbor/distanceAlgorithm/MinkowskiDistance.java
package mchehab.com.knearestneighbor.distanceAlgorithm;

public class MinkowskiDistance implements DistanceAlgorithm {

    private int p;

    public MinkowskiDistance(int p){
        this.p = p;
    }

    public int getP() {
        return p;
    }

    public void setP(int p) {
        this.p = p;
    }

    @Override
    public double calculateDistance(double x1, double y1, double x2, double y2) {
        double x = Math.pow(x1 - x2, p);
        double y = Math.pow(y1 - y2, p);
        double distance = Math.pow(x + y, 1/p);
        return distance;
    }
}

File: kotlin/src/main/java/mchehab/com/kotlin/distanceAlgorithm/MinkowskiDistance.kt
package mchehab.com.kotlin.distanceAlgorithm

import kotlin.math.pow

class MinkowskiDistance(var p: Int): DistanceAlgorithm {
    override fun calculateDistance(x1: Double, x2: Double, y1: Double, y2: Double): Double {
        val x = (x1 - x2).pow(p)
        val y = (y1 - y2).pow(p)
        val distance = (x1 + x2).pow(1.0/p)
        return distance
    }
}